grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractServerCall.java

```diff
@@ -673,6 +416,112 @@ private void invokeOnCancel() {
         }
     }
 
+    void onError(Throwable t) {
+        if (!closeCalled && !(t instanceof AbortedStreamException)) {
+            close(t);
+        }
+    }
+
+    @Nullable
+    final Listener<I> listener() {
+        return listener;
+    }
+
+    final void setListener(Listener<I> listener) {
+        checkState(this.listener == null, "listener already set");
+        this.listener = requireNonNull(listener, "listener");
+        invokeOnReady();
+    }
+
+    abstract void startDeframing();
+
+    @Nullable
+    final ResponseHeaders responseHeaders() {
+        return responseHeaders;
+    }
+
+    @Override
+    public void sendHeaders(Metadata metadata) {
+        if (ctx.eventLoop().inEventLoop()) {
+            doSendHeaders(metadata);
+        } else {
+            ctx.eventLoop().execute(() -> doSendHeaders(metadata));
+        }
+    }
+
+    private void doSendHeaders(Metadata metadata) {
+        if (isCancelled()) {
+            // call was already closed by a client or a timeout scheduler.
+            return;
+        }
+        checkState(responseHeaders == null, "sendHeaders already called");
+        checkState(!closeCalled, "call is closed");
+
+        final Compressor oldCompressor = compressor;
+        if (messageCompression && !clientAcceptEncoding.isEmpty()) {
+            final List<String> acceptedEncodings =
+                    ACCEPT_ENCODING_SPLITTER.splitToList(clientAcceptEncoding);
+            if (compressor != null) {
+                if (!acceptedEncodings.contains(compressor.getMessageEncoding())) {
+                    // resort to using no compression.
+                    compressor = Codec.Identity.NONE;
+                }
+            } else if (autoCompression) {
+                for (final String encoding : acceptedEncodings) {
+                    final Compressor compressor0 = compressorRegistry.lookupCompressor(encoding);
+                    if (compressor0 != null) {
+                        compressor = compressor0;
```

Comment:
This part was slightly changed from `setCompression(encoding)` while the master branch was merged.
`setCompression(encoding)` calls `responseFramer.setCompressor(ForwardingCompressor.forGrpc(compressor))` 
that is also invoked the end of this block.
/cc @ta7uw

Reply:
https://github.com/line/armeria/blob/1008ba517cf7dd9df5699d0ca62e3f451464d66c/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L267

Reply:
I checked the change of this part.
Looks good!